routes/requests.js

```diff
@@ -18,4 +20,63 @@ router.get('/', async (req, res) => {
   })
 })
 
+/**
+ * POST /requests
+ *
+ * Creates a new request and adds it to the collection.
+ * Requires the sender to be logged in.
+ *
+ * Request body: an request object
+ *
+ * @returns the newly created request object
+ */
+router.post('/', isLoggedIn, async (req, res) => {
+  const newRequest = new Request({
+    ...req.body,
+    requestor: req.session.user
+  })
+  await newRequest.save()
+  const request = await newRequest.populate(['item', 'itemOwner', 'requestor'])
+  res.send({
+    result: request
+  })
+})
+
+/**
+ * DELETE /requests/:id
+ *
+ * Removes an request from the collection.
+ *
+ * @param id: the request id to delete
+ * @returns the deleted request object
+ */
+router.delete('/:id', isLoggedIn, async (req, res) => {
+  const { id } = req.params
+
+  const deleted = await Request.findByIdAndDelete(id).populate(['item', 'itemOwner', 'requestor'])
+
+  res.send({
+    result: deleted
+  })
+})
+
+/**
+ * POST /requests/:id/accept
+ *
+ * Sets an items status to borrowed and adds it to the borrowed item history
+ * of the currently logged in user. Requires the sender to be logged in.
+ *
+ * @param id: the item id to borrow
+ * @return the updated item
+ */
+router.post('/:id/accept', isLoggedIn, async (req, res) => {
+  const { id: itemId } = req.params
+
+  const accepted = await Request.findByIdAndUpdate(itemId, { status: REQUEST_STATUS.ACCEPTED }, { new: true }).populate(['item', 'itemOwner', 'requestor'])
+
+  res.send({
+    result: accepted
+  })
+})
+
```

Comment:
Think these endpoints should only work if the sender is either the requestor (for delete) or the item owner (for delete/accept). 

Can you add some middleware functions, similar to isLoggedIn and isItemOwner, that check the user is authorized?

Reply:
@phreid could we do that as a potential improvement? Right now we still hit those cases because the front end filters for each page to only show a user pending requests if they have requested it, and only shows them received requests if they are the item owner.

Reply:
i'd like to have it in to keep it consistent with the rest of the API, but i can add it in later